### PR TITLE
Update Developer Guide: NFR, Glossary

### LINKED ISSUE #9: Update Developer Guide
Add the following information to the Developer Guide:

- Target user profile, value proposition, and user stories
- Use cases
- Non-functional requirements
- Glossary

### PR DESCRIPTION
**Pull request by yongping827**
Resolves #9.

Updated Non-Functional Requirements and Glossary.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -299,12 +299,19 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
```
Comment (wltan) [inline docs/DeveloperGuide.md:299]: Do glossary terms need to be italicized? May need to standardize.

Comment (yongping827) [reply docs/DeveloperGuide.md:299]: Glossary terms shall remain unformatted. Edits will be made.

**Review by ianyong — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -296,15 +296,22 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+1.  Should work on any mainstream OS as long as it has Java `11` or above installed.
 2.  Should be able to hold up to 1000 expenses/incomes without a noticeable sluggishness in performance for typical usage.
 3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4.  The data should be stored locally and should be in a human-editable text file.
+5.  The data should not be stored in a Database Management System (DBMS).
+6.  Should not require an installer.
+7.  Should not depend on any remote server.
 
 *{More to be added}*
```
Comment (wltan) [inline docs/DeveloperGuide.md:307]: Can I confirm that this line will be left unchanged? Otherwise LGTM.

**Review by zhaojj2209 — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (yongping827) [reply docs/DeveloperGuide.md:307]: Yes, it will remain unchanged.

Comment (wltan) [reply docs/DeveloperGuide.md:307]: Ok, we will review this again towards the end of v1.2.

**Review by wltan — APPROVED** [review]
LGTM

**Pull request merged**